bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/Utils.java

```diff
@@ -107,4 +126,79 @@ public static String generateId(String prefix, String path) {
         return StringUtils.join(prefix, ID_SEPARATOR, StringUtils.substring(DigestUtils.sha256Hex(path), 0, 10));
     }
 
+    public static Set<String> getPageResourceTypes(@NotNull Page page, @NotNull SlingHttpServletRequest request, @NotNull ModelFactory modelFactory) {
+        Set<String> resourceTypes = new HashSet<>();
+        resourceTypes.addAll(getResourceTypes(page.getContentResource(), request, modelFactory));
+        resourceTypes.addAll(getTemplateResourceTypes(page, request, modelFactory));
+        return resourceTypes;
+    }
+
+    @NotNull
+    public static Set<String> getResourceTypes(@NotNull Resource resource, @NotNull SlingHttpServletRequest request, @NotNull ModelFactory modelFactory) {
+        Set<String> resourceTypes = new HashSet<>();
+        resourceTypes.add(resource.getResourceType());
+        //resourceTypes.addAll(getSuperTypes(resource.getResourceType(), resolver));
```

Comment:
Should this code line commented out? If so, why not just remove it?

Reply:
Removed in d9f4bb2